pennylane_lightning/src/benchmarks/README.md

```diff
@@ -0,0 +1,260 @@
+          [--benchmark_context=<key>=<value>,...]
+          [--benchmark_time_unit={ns|us|ms|s}]
+          [--v=<verbosity>]
+```
+
+## Implementation details
+The `make gbenchmark` command compiles benchmark executables from the following files,
+- `Bench_BitUtil.cpp`,
+- `Bench_LinearAlgebra.cpp`,
+- `Bench_ApplyOperations.cpp`,
+- `Bench_ApplyMultiRZ.cpp`.
+
+
+### `benchmarks/utils`
+To benchmark the linear algebra and bit operations in PennyLane-Lightning, one can run:
+```Bash
+$ make gbenchmark
+$ ./BuildGBench/benchmarks/utils 
+```
+
+For example, the `std_innerProd_cmplx<T>` method in `Bench_LinearAlgebra.cpp` performs the benchmark for the inner product of two vectors with complex numbers (`std::complex<T>`) using `std::inner_product`. This is usual to try a few arguments in some ranges and generate a benchmark for each such value. The GB offers `Range` and `Ranges` to do so, 
+```C
+BENCHMARK(std_innerProd_cmplx<float>)
+    ->Range(1l << 5, 1l << 10);
+```
+
+By default the arguments in the range are generated in multiples of 8 and the command above picks `{32, 64, 512, 1024}`. To change the range multiplier, one should use `RangeMultiplier`. The following code selects `{4, 8, 16, 32, 64, 128, 256, 512, 1024}`,
+```C
+BENCHMARK(std_innerProd_cmplx<float>)
+    ->RangeMultiplier(2)
+    ->Range(1l << 2, 1l << 10);
+```
+
+To filter the benchmark results to run only `std_innerProd_cmplx`, one can run:
+```Bash
+$ ./BuildGBench/benchmarks/utils --benchmark_filter=std_innerProd_cmplx
+```
+
+Besides, one can use `--benchmark_time_unit` to get the results in `{ns|us|ms|s}` too. Check **GB CLI Flags** for the list of flags. 
+
+
+### `benchmarks/apply_operations`
+To benchmark the `Pennylane::StateVectorManaged` and `applyOperation` in PennyLane-Lightning, one can run:
+```Bash
+$ make gbenchmark
+$ ./BuildGBench/benchmarks/apply_operations
+```
+
+The follwoing arguments could be altered:
+- `Pennylane::Gates::KernelType` 
+- The list of gates that script would randomly pick from to apply to the state vector. 
+- The range for the number of gates to be applied.
+- The range for the number of qubits. 
+
+For example, in the code below, 
+- `applyOperationsFromRandOps` is the name of the method in `Bench_ApplyOperations.cpp`, 
+- `LM_RXYZ` is just a name for the parameters, 
+- `Pennylane::Gates::KernelType::LM` is the kernel type,
+- `{"RX", "RY", "RZ"}` is the list of gates,
+- `{8, 64}` is a range for the number of gates to be randomly picked from the list of gates and be applied, and 
+- `{4, 24}` is a range for the number of qubits.  
+
+```C
+BENCHMARK_CAPTURE(applyOperationsFromRandOps, LM_RXYZ,
+                  Pennylane::Gates::KernelType::LM, {"RX", "RY", "RZ"})
+    ->RangeMultiplier(2)
+    ->Ranges({{8, 64}, {4, 24}});
+```
+
+```Bash
+-----------------------------------------------------------------------------------
+Benchmark                                         Time             CPU   Iterations
```

Comment:
Is it possible to get cache/core information out of this too, or is it just the overall runtime numbers?

Reply:
The benchmark suite always reports cache/core information too; I will include the info to the README file. Also, regarding "Time", You can also get the real time by adding `->UseRealTime()` to the end of any benchmark calls.